### DISCUSSION TITLE
Change the default language to Chinese

### DISCUSSION THREAD

**Discussion by c2006878** [nocodb/nocodb#6895] [General]
The default language now is English. If I want to change it to Chinese, which configuration file do I need to modify? Not just the switching of buttons.

**Comment by dstala** [comment]
@c2006878 Currently, there is no provision to do that. You will have to switch language manually by using `Language` menu as seen in the image below

![Screenshot 2023-11-06 at 10 21 19 AM](https://github.com/nocodb/nocodb/assets/86527202/bfcb6545-dcb9-4a14-ba92-e367d49a7af2)

**Comment by c2006878** [comment]
I checked the information and found that the configuration of i18n can be modified in packages/nc gui/plugins/a.i18n.ts. But it didn't take effect, why is that?

export const createI18nPlugin = async (): Promise<NocoI18n> =>
  createI18n({
    locale: 'en', // Set the initial locale

    fallbackLocale: 'en', // Set the fallback locale in case the current locale can't be found

    legacy: false, // disable legacy API (we use the composition API and inject utilities)

    globalInjection: true, // enable global injection, so all utilities are injected into all components
  })

**Comment by githubmyname** [comment]
There are a lot of locations in the translation file to do the translation, why gui display incomplete? Only part of the translation is shown.
![截图_选择区域_20231106225208](https://github.com/nocodb/nocodb/assets/149666360/4172a504-752e-4fcb-81ed-7ebc5478342a)
![截图_选择区域_20231106225217](https://github.com/nocodb/nocodb/assets/149666360/60990160-afa6-4807-810d-65409b6f129b)
![截图_选择区域_20231106225251](https://github.com/nocodb/nocodb/assets/149666360/dca5662c-86c3-4a2e-8ea3-ec0d3cd77aed)

**Comment by wingkwong** [comment]
> I checked the information and found that the configuration of i18n can be modified in packages/nc gui/plugins/a.i18n.ts. But it didn't take effect, why is that?
> 
> export const createI18nPlugin = async (): Promise => createI18n({ locale: 'en', // Set the initial locale
> 
> ```
> fallbackLocale: 'en', // Set the fallback locale in case the current locale can't be found
> 
> legacy: false, // disable legacy API (we use the composition API and inject utilities)
> 
> globalInjection: true, // enable global injection, so all utilities are injected into all components
> ```
> 
> })

@c2006878 

If you are talking about changing the default locale locally, your approach is correct. You may need to update the following files as well. Just search `en` and replace the corresponding value. 

<img width="432" alt="image" src="https://github.com/nocodb/nocodb/assets/35857179/37794645-8e6b-46df-b060-84a3b0845ca0">

Moreover, since `lng` is read from local storage, after modifying the source code, you need to clear the local storage first. 

![image](https://github.com/nocodb/nocodb/assets/35857179/5fe47cb0-14d5-45d1-b264-9b74d0b6c69c)

**Comment by wingkwong** [comment]
@githubmyname that means the translation hasn't been applied for that particular text.

**Comment by dstala** [comment]
Thanks for the details @wingkwong 
Archiving this thread as a discussion to help other users with similar requirements.

**Comment by githubmyname** [comment]
> @githubmyname that means the translation hasn't been applied for that particular text.

What do I need to do to apply it? Or do you need to wait for updates from developers? thank.

**Reply by wanlidan** [reply]
Hello, I'm also having the same issue. Have you managed to resolve it?

**Comment by genan1989** [comment]
Seeking relevant solutions. Mixing Simplified Chinese and English is very uncomfortable
